.claude/commands/find-component-pattern.md

```diff
@@ -0,0 +1,651 @@
+- Useful for identifying technical debt and refactoring opportunities
+- Can identify opportunities for creating shared wrapper components
+- Output can inform component API design decisions
+- Run periodically as part of code quality audits
+- Particularly valuable before making breaking changes to a component's API
+- Helps new team members understand how components are typically used
+- **Use `--show-recommendations`** when you need actionable guidance on fixing issues and standardizing patterns
+- **Skip recommendations** (default) when you only need to understand current usage patterns without prescriptive advice
+
+---
+
+## Example Workflows
+
+### Example 1: Basic Usage - User Declines Recommendations
+
+**User:** `/find-component-pattern`
+**Assistant:**
+```
+Which component would you like me analyze?
+
+Use --scope=SCOPE to limit scope (optional):
+- all - Search entire frontend (default)
+- core - Only core components (frontend/public/components/)
+- package:NAME - Specific package (e.g., package:dev-console)
+```
+**User:** "ResourceLink"
+
+**Assistant proceeds:**
+1. **Phase 1:** Finds ResourceLink component definition, searches all frontend code for 87 usages
+2. **Phase 2:** Groups into 3 patterns, identifies 4 outliers
+3. **Phase 3:** Generates report with patterns, outliers, and statistics
+4. **Phase 4:** Outputs text prompt:
+   ```
+   Would you like to see recommendations for standardizing patterns and fixing inconsistencies? (yes/no)
+   ```
+
+**User:** "no"
+
+**Output:** Analysis showing Pattern 1 (passing groupVersionKind, name, namespace) is used in 70% of cases, Pattern 2 (passing kind, name, namespace) in 25%, Pattern 3 (minimal props) in 5%, plus 4 outliers missing required props. Analysis ends without recommendations.
+
+---
+
+### Example 2: Basic Usage - User Accepts Recommendations
+
+**User:** `/find-component-pattern`
+**Assistant:**
+```
+Which component would you like me analyze?
+
+Use --scope=SCOPE to limit scope (optional):
+- all - Search entire frontend (default)
+- core - Only core components (frontend/public/components/)
+- package:NAME - Specific package (e.g., package:dev-console)
+```
+**User:** "ResourceLink"
+
+**Assistant proceeds:**
+1. **Phase 1:** Finds ResourceLink component definition, searches all frontend code for 87 usages
+2. **Phase 2:** Groups into 3 patterns, identifies 4 outliers
+3. **Phase 3:** Generates report with patterns, outliers, and statistics
+4. **Phase 4:** Outputs text prompt:
+   ```
+   Would you like to see recommendations for standardizing patterns and fixing inconsistencies? (yes/no)
+   ```
+
+**User:** "yes"
+
+**Output:** Full analysis with patterns, outliers, statistics, AND prioritized recommendations for standardization.
+
+---
+
+### Example 3: With Scope (Core Components Only)
+
+**User:** `/find-component-pattern --scope=core`
+**Assistant:**
+```
+Which component would you like me analyze?
+
+Use --scope=SCOPE to limit scope (optional):
```

Comment:
Here the scope is already define, so the user should just specify the component name, right ?